pyincore/analyses/waterfacilityrestoration/waterfacilityrestoration.py

```diff
@@ -0,0 +1,179 @@
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License v2.0 which accompanies this distribution,
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+
+
+"""
+Water Facility Restoration
+"""
+
+import numpy as np
+
+from pyincore import BaseAnalysis, RestorationService
+from pyincore.models.restorationcurveset import RestorationCurveSet
+
+
+class WaterFacilityRestoration(BaseAnalysis):
+    """Computes water facility restoration for an earthquake, tsunami, tornado, or hurricane exposure.
+
+    """
+
+    def __init__(self, incore_client):
+        self.restorationsvc = RestorationService(incore_client)
+        super(WaterFacilityRestoration, self).__init__(incore_client)
+
+    def run(self):
+        """Performs Water facility restoration analysis by using the parameters from the spec
+        and creates an output dataset in csv format
+
+        Returns:
+            bool: True if successful, False otherwise
+        """
+        mapping_set = self.get_input_dataset("dfr3_mapping_set")
+
+        # Get Fragility key
+        restoration_key = self.get_parameter("restoration_key")
+        if restoration_key is None:
+            restoration_key = "Restoration ID Code"
+
+        end_time = self.get_parameter("end_time")
+        if end_time is None:
+            end_time = 365.0
+
+        time_interval = self.get_parameter("time_interval")
+        if time_interval is None:
+            time_interval = 1  # 1 day
```

Comment:
maybe rename this variable so it's more clear that the time interval is in days? If not, the comment doesn't really make the code more readable

Reply:
oh maybe i'll put the day in the input parameter description section in the spec?
this could be used for any time interval; it's just a convention that it's using day